src/app/public/modules/split-view/split-view.component.ts

```diff
@@ -0,0 +1,423 @@
+
+import {
+  SkySplitViewMediaQueryService
+} from './split-view-media-query.service';
+
+import {
+  SkySplitViewListComponent
+} from './split-view-list.component';
+
+import {
+  SkySplitViewWorkspaceComponent
+} from './split-view-workspace.component';
+
+let nextId = 0;
+
+@Component({
+  selector: 'sky-split-view',
+  templateUrl: './split-view.component.html',
+  styleUrls: ['./split-view.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: [
+    trigger(
+      'blockAnimationOnLoad', [
+        transition(':enter', [])
+      ]
+    ),
+    trigger(
+      'listEnter', [
+        transition(`void => *`, [
+          style({transform: 'translate(-100%)'}),
+          animate('150ms ease-in')
+        ])
+      ]
+    ),
+    trigger(
+      'workspaceEnter', [
+        transition(`void => *`, [
+          style({transform: 'translate(100%)'}),
+          animate('150ms ease-in')
+        ])
+      ]
+    )
+  ]
+})
+export class SkySplitViewComponent implements OnInit, AfterViewInit, OnDestroy {
+
+  public appSettings: any;
+
+  public splitViewId: string = `sky-split-view-${++nextId}`;
+
+  // Max needs to start as something to allow input range to work.
+  // This value is updated as soon as the user takes action.
+  public listWidthMax = 9999;
+
+  public listWidthMin = 100;
+
+  public listWidthDefault = 320;
+
+  public iteratorNextButtonDisabled = false;
+
+  public iteratorPreviousButtonDisabled = false;
+
+  @Output()
+  public iteratorNextButtonClick = new EventEmitter<void>();
+
+  @Output()
+  public iteratorPreviousButtonClick = new EventEmitter<void>();
+
+  public set isListVisible(value: boolean) {
+    this._listVisible = value;
+  }
+
+  public get isListVisible() {
+    return !this.isMobile || this._listVisible;
+  }
+
+  public get workspaceVisible() {
+    return !this.isMobile || !this._listVisible;
+  }
+
+  @Input()
+  public set listWidth(value: number) {
+    if (value) {
+      this._listWidth = value;
+      this.updateBreakpoints();
+    }
+  }
+  public get listWidth() {
+    if (this.isMobile) {
+      return undefined;
+    } else {
+      if (this._listWidth > this.listWidthMax) {
+        return this.listWidthMax;
+      } else if (this._listWidth < this.listWidthMin) {
+        return this.listWidthMin;
+      } else {
+        return this._listWidth || this.listWidthDefault;
+      }
+    }
+  }
+
+  @Input()
+  public messageStream = new Subject<SkySplitViewMessage>();
+
+  @Input()
+  public ariaRole: string;
+
+  @Input()
+  public ariaDescribedBy: string;
+
+  @Input()
+  public ariaLabelledBy: string;
+
+  public isDragging = false;
+
+  @Output()
+  public beforeWorkspaceClose = new EventEmitter<SkySplitViewBeforeWorkspaceCloseHandler>();
+
+  public nextButtonDisabled = false;
+
+  public previousButtonDisabled = false;
+
+  public isMobile = false;
+
+  private _listWidth: number;
+
+  private xCoord = 0;
+
+  private _listVisible = true;
```

Comment:
The order of the properties is a bit off.
https://docs.blackbaud.com/skyux-team-docs/style#class-properties

Reply:
Updated!